### ISSUE TITLE
How do you connect different models using Ray.

### ISSUE THREAD

**Issue by zpcalan** [OpenRLHF/OpenRLHF#472]
For example, reward model has 8 GPU cards with TP and DP configuration. Actor model might have TP&PP&DP(just for example) occupying 64 GPUs.
How do you connect Actor's last stage's output to reward model while using Ray? Or how do you express specific  communication relationships?
I used Ray once but I don't think they could handle or express the communication between one [Device List] to another [Device List] because the outputs and inputs might be on different devices according to different parallel configs.
I really hope you can help me out with this question. Thank you!
: )

**Comment by hijkzzz** [comment]
Ray can handle this.
Please read our ray-related codes carefully.
Thanks~

**Comment by zpcalan** [comment]
@hijkzzz Yes, we are trying to deploy your framework on our devices lately but we still can not find the code about our question. Would you mind if you post the related code link in the comment?
We will be very appreciated. : )

**Comment by tsaoyu** [comment]
On high level, Ray is passing those messages around so you don't have to explicitly express physical data flow of those messages by hand. 

For your curiosity, you can check 
https://github.com/OpenRLHF/OpenRLHF/blob/main/openrlhf/trainer/ray/ppo_actor.py#L187 for Actor Model definition, and https://github.com/OpenRLHF/OpenRLHF/blob/main/openrlhf/trainer/ray/launcher.py#L106 for Reward Model definition. Once they decorated with `@ray.remote(num_gpus=1)` they become Ray Actors that the underlying communication go through Ray. 

https://github.com/OpenRLHF/OpenRLHF/blob/main/openrlhf/cli/train_ppo_ray.py#L162

Hope that is a good start point for you to understand the data flow in OpenRLHF.
